pkg/edit/repl.go

```diff
@@ -0,0 +1,51 @@
+package edit
+
+// This file encapsulates functionality related to a complete REPL cycle. Such as capturing
+// information about the most recently executed interactive command.
+
+import (
+	"src.elv.sh/pkg/eval"
+	"src.elv.sh/pkg/eval/vals"
+	"src.elv.sh/pkg/eval/vars"
+)
+
+//elvdoc:var after-command
+//
+// A list of functions to call after each interactive command completes. There is one pre-defined
+// function used to populate the [`$edit:command-duration`](./edit.html#editcommand-duration)
+// variable. Each function is called with a single pseudo-map argument containing the following
+// keys:
+//
+// * `command`: A string containing the code that was executed.
+//
+// * `duration`: A [floating-point number](https://elv.sh/ref/language.html#number) representing the
+// command execution duration in seconds.
+//
+// * `error`: An [exception](../ref/language.html#exception) object if the command terminated with
+// an exception, else [`$nil`](../ref/language.html#nil).
+//
+// @cf edit:command-duration
+
+//elvdoc:var command-duration
+//
+// Duration, in seconds, of the most recent interactive command. This can be useful in your prompt
+// to provide feedback on how long a command took to run. The initial value of this variable is the
+// time to evaluate your *~/.elvish/rc.elv* script before printing the first prompt.
+//
+// @cf edit:after-command
+
+var CommandDuration float64
```

Comment:
This should not be a global variable.

Reply:
Agreed. That it is global is an artifact of a much earlier version of this change.